### DISCUSSION TITLE
Azote 1.12.8/1.12.9 (2024-05-28)

### DISCUSSION THREAD

**Discussion by nwg-piotr** [nwg-piotr/nwg-shell#337] [Changelog]
- added `azote` entry point, hardcoded /usr/bin/azote file removed; should hopefully close https://github.com/nwg-piotr/azote/issues/189;
- in 1.12.9 fixed images path, previously broken in 1.12.8.

**Comment by 01micko** [comment]
Just a note that `azote-1.12.9` builds and runs fine on debian/trixie(testing) for anyone interested. I built a `.deb` using `debmake` and creating a custom `install` file and editing `control`, `rules`, and associated files then ran `debuild` but you can use whichever debian tool you like. It certainly needs improvement as there are numerous build/install warnings at the `lintian` stage.

I'll post the `install`, `control` and `rules` files. The rest you can edit as needed. I deleted `debian/tests/`. Take note that this is `wayland` specific and other `Depends` would be needed to work on `X11`.  All the `Build-Depends` are required.

#### debian/control

```
Source: azote
Section: misc
Priority: optional
Maintainer: Mick Mars <01micko@gmail.com>
Build-Depends: debhelper-compat (= 13), python3-setuptools, python3-build, python3-installer, python3-wheel, python3-all, dh-python
Standards-Version: 4.6.1
Homepage: https://github.com/nwg-piotr/azote
Rules-Requires-Root: no

Package: azote
Architecture: any
Multi-Arch: foreign
Depends: ${misc:Depends}, ${shlibs:Depends}, python3, python3-pil, python3-send2trash, python3-cairo, libgtk-3-0t64, swaybg, wlr-randr
Description: This is a GTK+3 - based picture browser and background setter
 The user interface is being developed with multi-headed setups in mind.
 Azote also includes several colour management tools.
```

#### debian/rules

```make
#!/usr/bin/make -f
export PYBUILD_NAME=azote

%:
	dh $@ --with python3 --buildsystem=pybuild
```

#### debian/install 

this is a file where you make it yourself

```
dist/azote.svg usr/share/pixmaps
dist/azote.desktop usr/share/applications
dist/indicator* usr/share/azote
```

That's it! Happy building. (edited to add _debian/install_)

**Reply by nwg-piotr** [reply]
I think I'll add a link to this post to the Azote README. Many thanks!

**Reply by 01micko** [reply]
Ok, I've worked around the bug where `azote.desktop` and friends in `dist` are not installed. Stupid debian quirk methinks because if I patch `setup.py` like so; `s/find_packages/find_namespace_packages/g` the AUR works in debian so I did that in the recipe and they still fail to install! Life was so much simpler when I was a Slacker! Slackware, even though it's old school 'just works' ootb.
Anyway, the trick is to use a `debian/install` file with relpaths. All works well now, so I'll edit my above post with that detail.

**Reply by nwg-piotr** [reply]
Very well, the link will be still the same. BTW: all my stuff is available on Slackware, thanks to @mac-a-r0ni.

**Reply by 01micko** [reply]
Thanks :)
